capstone/src/main/java/com/google/sps/data/ReadData.java

```diff
@@ -0,0 +1,71 @@
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+
+public class ReadData {
+  final String DELIMITER = ",";
+
+  public void readCSV() {
+    BufferedReader reader = null;
+    FileReader file = null;
+    DatastoreService datastoreService = DatastoreServiceFactory.getDatastoreService();
+
+    try {
+      file = new FileReader("assets/surveyresponse.csv");
+      reader = new BufferedReader(file);
+      String header = reader.readLine();
+      String line = "";
+
+      while ((line = reader.readLine()) != null) {
+        String[] values = line.split(DELIMITER);
+        String id = values[0];
+        double score = Double.parseDouble(values[2]);
+        String gender = values[3];
+        String ageRange = values[4];
+        long responseTime = Long.parseLong(values[5]);
+        String date = values[6];
+        long day = Long.parseLong(values[7]);
+
+        Entity entity = new Entity("Response", id);
+        Entity inStore; // Entity that may or may not exist in Datastore
+
+        try {
+          inStore = datastoreService.get(entity.getKey());
+        } catch (EntityNotFoundException e) {
+          inStore = null;
+        }
+
+        if (inStore == null) {
+          entity.setProperty("id", id);
+          entity.setProperty("score", score);
+          entity.setProperty("gender", gender);
+          entity.setProperty("ageRange", ageRange);
+          entity.setProperty("responseTime", responseTime);
+          entity.setProperty("date", date);
+          entity.setProperty("day", day);
+
+          datastoreService.put(entity);
+        }
+      }
+    } catch (FileNotFoundException e) {
+      e.printStackTrace();
```

Comment:
You don't need to print stack trace. It should print the stack trace automatically.

Reply:
What action should in the block?

Reply:
You don't need to take action. If file is not found, you'd want to throw an exception right? That should happen automatically without needing a try catch block.

In the future, when you rely on the API however, you probably want a try catch block to make sure you have a proper xlsx file being sent.

Reply:
The compiler complains that those exceptions are not caught, would I throw the exceptions in the method signature?